openmc/geometry.py

```diff
@@ -382,6 +389,24 @@ def get_all_surfaces(self):
             surfaces = cell.region.get_surfaces(surfaces)
         return surfaces
 
+    def get_redundant_surfaces(self):
+        """Return all of the topologically redundant surface ids
+
+        Returns
+        -------
+        dict
+            Dictionary whose keys are the ID of a redundant surface and whose
+            values are the topologically equivalent :class:`openmc.Surface`
+            that should replace it.
+
+        """
+        tally = defaultdict(list)
+        surfaces = self.get_all_surfaces()
+        for sid, surf in surfaces.items():
+            tally[surf.description].append(sid)
```

Comment:
All we really need for the key here is something that uniquely identifies the surface. To me, this `description` attribute is kind of overkill for what's needed. My suggestion would be to have the key just be a tuple containing the type of the surface and the coefficients. Something like:
```Python
key = (surf._type,) + tuple(surf._coefficients[k] for k in surf._coeff_keys)
```
As an example, for a sphere with radius 3.0, this would produce the tuple `('sphere', 0.0, 0.0, 0.0, 3.0)`. I don't think we really need boundary info because if your geometry has two equivalent surfaces with _different_ boundary conditions, you're really in for trouble already.